src/libultra/io/cartrominit.c

```diff
@@ -1,3 +1,63 @@
 #include "global.h"
 
-#pragma GLOBAL_ASM("asm/non_matchings/boot/cartrominit/osCartRomInit.s")
+OSPiHandle CartRomHandle;
+
+OSPiHandle* osCartRomInit(void) {
+    register u32 a;
+    register s32 status;
+    register u32 prevInt;
+    register u32 lastLatency;
+    register u32 lastPageSize;
+    register u32 lastRelDuration;
+    register u32 lastPulse;
+
+    static u32 D_800980D0 = 1;
+
+    __osPiGetAccess();
+
+    if (!D_800980D0) {
+        __osPiRelAccess();
+        return &CartRomHandle;
+    }
+
+    D_800980D0 = 0;
+    CartRomHandle.type = DEVICE_TYPE_CART;
+    CartRomHandle.baseAddress = 0xB0000000;
+    CartRomHandle.domain = 0;
+    CartRomHandle.speed = 0;
+    bzero(&CartRomHandle.transferInfo, sizeof(__OSTranxInfo));
+
+    /* Uses `status & PI_STATUS_ERROR` in OoT */
+    while (status = HW_REG(PI_STATUS_REG, u32), status & (PI_STATUS_BUSY | PI_STATUS_IOBUSY)) {
```

Comment:
This is a mistake in oot. `PI_STATUS_ERROR` is currently wrongly defined in oot, and if it were defined correctly it would not be usable here, so I don't think it's worth mentioning.

Reply:
It's still different from OoT, though... is it a libultra version difference?

Reply:
`PI_STATUS_ERROR` in oot is currently 0x3, so 0x2 | 0x1, or `PI_STATUS_BUSY | PI_STATUS_IOBUSY`, which is the same as here. Unless I've misunderstood and you're referring to something else?

Reply:
I copypasted the OoT file, which used 0x4 here instead of 0x3, so seems to be a genuine code difference.

Reply:
OoT definitely uses 0x3 here too is what I'm trying to say. It's just our definition if `PI_STATUS_ERROR` is not the same as in OoT since it's wrong there currently (but I fixed it in the handwritten asm PR)

Reply:
Oh right, I get it now.